### DISCUSSION TITLE
[BP#38] Utiliser Ajax pour certaines zones de contenu

### DISCUSSION THREAD

**Discussion by Fred-artwai** [cnumr/best-practices#236] [Bonnes pratiques] [answered]
Le mot AJAX est peut-être un peu daté.
J'utiliserais le mot "requête asynchrone".

Ce qui me parait important çà serait de préciser les technos qui permettent de le faire. Et là on va au delà de l'AJAX.
- requête asynchrone dans un js classique
- mais aussi service worker
- et pourquoi pas un web component.

**Comment by tbroyer** [comment, marked as answer]
> Ce qui me parait important çà serait de préciser les technos qui permettent de le faire.

Dans le cadre de ces bonnes pratiques, je ne suis pas sûr.

L'idée de cette BP est essentiellement de dire: si c'est possible, il vaut mieux privilégier des pages "incomplètes" mais qui peuvent être mises en cache, et qui seront "complétées" avec du contenu récupéré _via_ du JS. Ça peut s'appliquer aux commentaires sur un article par exemple: ils changent bien plus que l'article, donc ils peuvent être récupérés séparément pour permettre à la page d'être mise en cache indépendamment (et dans le cas des commentaires, on pourra les récupérés uniquement "à la demande")
On pourrait penser aux "trending topics" aussi, etc.
Au lieu de générer côté serveur des "petites" parties (généralement annexes) de la page, et donc empêcher sa mise en cache (et générer ainsi pas mal de trafic et de calculs sur le serveur), mieux vaut les récupérer côté client.

En d'autres termes: https://jasonformat.com/islands-architecture/

**Reply by ACTLEM** [reply]
J'ai créé une issue pour modifier le titre et reformuler l'idée de la BP comme expliqué par @tbroyer : https://github.com/cnumr/best-practices/issues/291
